src/basic/macro.h

```diff
@@ -103,74 +103,6 @@ static inline size_t GREEDY_ALLOC_ROUND_UP(size_t l) {
                 (type*)( (char *)UNIQ_T(A, uniq) - offsetof(type, member) ); \
         })
 
-/* Logging for various assertions */
```

Comment:
you moved this here just a few commits earlier in the same PR. would prefer if these would be moved to the right place right away.. (and as mentioned I think the right place would be some assert-util.[ch] or so

Reply:
See above, need to take circular dependencies into account.

Reply:
Fixed